isamples_metadata/taxonomy/metadata_models.py

```diff
@@ -0,0 +1,305 @@
+        if "IODP" in field_to_value["cruiseFieldPrgrm"] or \
+                "ODP" in field_to_value["cruiseFieldPrgrm"]:
+            if "core" in field_to_value["sampleType"].lower():
+                return "Mixture of sediment and rock"
+            if field_to_value["sampleType"] == "Individual Sample":
+                return "Sediment or Rock"
+            if "macrofossil" in field_to_value["description"].lower():
+                return "Rock"
+        if "dredge" in field_to_value["sampleType"].lower():
+            return "Natural Solid Material"
+        if field_to_value["primaryLocationType"] == "wetland" and \
+                "core" in field_to_value["sampleType"].lower():
+            return "Material"
+        if field_to_value["sampleType"] == "U-channel":
+            return "Sediment"
+        if field_to_value["sampleType"] == "CTP":
+            return "Liquid"
+        if field_to_value["sampleType"] == "Individual Sample>Cylinder":
+            return "Material"
+        # if the record cannot be classified by the defined rules
+        return None
+
+    def classify_by_rule(self, text: str, description_map: dict):
+        """ Checks if the record can be classified by rule
+        If the record corresponds to a rule, returns the rule-defined label
+        Else return None
+        """
+        # 1. check if record does not have enough information
+        # if not enough information given,
+        # give "Material" label
+        if not self.check_informative(text, description_map):
+            return "Material"
+
+        # 2. rule-based classification
+        # extract fields that we need to consider for the rules
+        fields_to_check = [
+            "sampleType",
+            "cruiseFieldPrgrm",
+            "igsnPrefix",
+            "description",
+            "primaryLocationType"
+        ]
+        # build a map that stores the fields that we are interested
+        field_to_value = collections.defaultdict(str)
+        for key, value in description_map.items():
+            if key in fields_to_check:
+                field_to_value[key] = value
+
+        # check if record is invalid
+        # i.e., not a sample
+        if self.check_invalid(field_to_value):
+            return "Invalid"
+
+        # check if the fields fall into the rules
+        # if it does not, return None
+        result = self.classify_by_sample_type(field_to_value)
+        if result:
+            # map to controlled vocabulary
+            return SESARClassifierInput.source_to_CV[result]
+        else:
+            return None
+
+    def classify_by_machine(self, text: str):
+        """ Returns the machine prediction on the given
+        input record
+        """
+        prediction, prob = self._model.predict(text)
+        return (
+            SESARClassifierInput.source_to_CV[prediction],
+            prob
+        )
+
+    def predict_material_type(
+        self, source_record: dict
+    ) -> Tuple[str, float]:
```

Comment:
This is a bit of a nit, but it would be nicer if this returned a simple class like `PredictionResult` that looked something like:

```
class PredictionResult:
    def __init__(self, value: str, confidence: float):
        self.value = value
        self.confidence = confidence
```

this way, callers won't need to go look at the code that's returning it to figure which index in the tuple is the value and which is the confidence.  Just makes it a little easier for people to work with.

Reply:
It looks like all the of different prediction methods return roughly the same thing, so we could use the same class everywhere.